### PR TITLE
Feature/various devices

### PR DESCRIPTION
**Pull request by ouberlord**
A small variety of custom device-types and module-types that I had to create for my environment that I'd like to contribute back to the community repository.

### CODE REVIEW

**Comment by danner26** [comment]
Hello, we have released a new set of tests and features to assist with properly creating definition files for the DeviceType-Library. These tests will ensure that all files conform to a standardized format and contain valid data. Over the coming weeks we will be adding more tests to handle other currently undefined cases. If you notice anything that you believe is a bug, please @ me in a comment and details what you think is incorrect.

Along with additional test cases we have added a feature that will annotate why tests fail. To access these annotations after a test has completed, please navigate to the _Files changed_ tab within your PR. At the very bottom of the page you will see a list of errors along with the filename in question and the error which was found.

With all this being said, if your PR is validated then great! Please @ me me so I can review it further! If your PR is not valid, please review the issues and correct them; in the meantime I will mark your PR as a draft. After correcting the issues and recieving a valid status, please mark the PR as ready to be reviewed. Thank you!

**Comment by danner26** [comment]
Hello @ouberlord, today we have added additional test cases to verify that devices are defined properly from a power standpoint. Please review the failure messages and correct any issues. Once that is complete and all checks are passing, you may mark this PR ready for review. Thanks!

**Comment by ouberlord** [comment]
I am still seeing the same failure message from, before, but I am not seeing where or why in particular it is failing. I see the summary line of "Error: Process completed with exit code 1.", but not where the process is encountering an issue that would result in such. What am I overlooking?

**Comment by danner26** [comment]
Weirdly enough, I see what you mean. I'd like to try to trace this down further. Could you give me access to your fork so I can push to it and run some tests?

**Comment by ouberlord** [comment]
I wasn't sure how to grant you access to the fork (I'm pretty new to GitHub) but I added you as a collaborator to my repository that contains the branch I was working with. If that isn't right, let me know where I missed what you had in mind.

**Comment by danner26** [comment]
That was exactly what I needed, thank you!
I will get back to you soon regarding what is going on

**Comment by danner26** [comment]
Okay so I figured out what was going on. In the validation pipeline there are multiple steps that lead up to the final pytest steps we have been improving lately. The first 3 steps that deal with formatting and linting YAML files dont have the best logging. I have taken an action to attempt to improve that.

The first issue, which specifically caused the process error, was due to multiple files which were improperly formatted. The Format YAML files step corrected the issues, which causes the pipeline to fail (which is expected). The origin files which I have fixed were:

- device-types/APC/AP8832.yaml
- device-types/F5/BIG-IP-i7800.yaml
- device-types/Tripp Lite/RS-0615-F.yaml
- module-types/NetApp/IOM12.yaml
- module-types/NetApp/PSU.yaml

I have gone ahead and fixed these, and pushed them into your branch.

**Comment by ouberlord** [comment]
Ah, awesome. I saw the new info in the files changed tab, so I made the necessary changes. However, now the test is failing in the same non-specific way as before.

**Comment by danner26** [comment]
So this is now failing on the yaml formatting step, which means one or many files are improperly formatted. The easiest way to troubleshoot this is to locally run `pre-commit run --config .pre-commit-hooks-config.yaml --all-files` and `pre-commit run --config .pre-commit-yamlfmt-config.yaml --all-files`

Doing so reveals that DS224C and DS460C are imporoperly formatted. The main issue is the your position elements are suppose to be strings. They are currently ints, so we need to wrap them in single quotes (`'`). I have gone ahead and fixed this for you. Now you should see the remaining errors.

**Comment by ouberlord** [comment]
I fixed what I was aware of, but am hitting issues with the latest. I'm also having issues running those commands; with the second one its ignoring the slashes in the path, and with the third It's missing "pytest" and I'm not sure how to proceed.
image

**Comment by danner26** [comment]
I have fixed the formatting errors. You can review my commits if youd like

Please check my comments and they are still valid. Also please fix any test errors as well. Feel free to mark this PR as ready to review after all checks pass

File: device-types/Fortinet/FG-60D.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ manufacturer: Fortinet
 model: FortiGate 60D
 slug: fortinet-fg-60d
 part_number: FG-60D
-u_height: 0
+u_height: 1
```
[the remark this answers is not in the document]
Comment (danner26) [reply device-types/Fortinet/FG-60D.yml]: Resolved for you.

**Review by danner26 — CHANGES_REQUESTED** [review]
I have resolved as many of the issues as I can. Please fix the power missing issue with the FAS devices. The chassis usually houses the power which powers the children devices.

Also, please check the NetApp PSU. `Power Supply Unit` is not a valid model

**Pull request closed without merging**